pennylane/devices/default_qubit.py

```diff
@@ -311,7 +311,7 @@ def _apply_operation(self, state, operation):
         if operation.__class__.__name__ in self._apply_ops:
             shift = int(self._ndim(state) > self.num_wires)
             axes = [ax + shift for ax in self.wires.indices(wires)]
-            return self._apply_ops[operation.base_name](state, axes)
+            return self._apply_ops[operation.__class__.__name__](state, axes)
```

Comment:
are the changes here and in null_qubit some kind of merge with #4130 ? the changes are not the same so I'm unsure

Reply:
Oh I didn't realize there was another PR changing these lines. I believe the changes in #4130 would also make this work, as we're addressing the same issue. The main problem here was checking that `operation.__class__.__name__` was in the keys and then indexing with `operation.base_name`, so I changed it to be consistent with the `if` statement.